server/src/main/java/org/elasticsearch/index/mapper/SourceFieldMapper.java

```diff
@@ -105,7 +104,7 @@ public SourceFieldMapper build(BuilderContext context) {
         public static final SourceFieldType INSTANCE = new SourceFieldType();
 
         private SourceFieldType() {
-            super(NAME, false, false, TextSearchInfo.NONE, Collections.emptyMap());
+            super(NAME, false, false, true, TextSearchInfo.NONE, Collections.emptyMap());
```

Comment:
I think we need to pass `false` here if `enabled` is `false`?

Reply:
yes I meant to get back to this, I was not sure either. I looked at it from the perspective of "if enabled there is no _source", but actually _source is still mapped and stored should be set to false. will update.

Reply:
and btw, I am setting doc_values to true rather than stored :D